models/Message.model.js

```diff
@@ -0,0 +1,25 @@
+const mongoose = require("mongoose");
+const { Schema, model } = mongoose;
+
+const messageSchema = new Schema(
+    {
+        destination: {
+            type: String
+        },
+        message: {
+            type: String
+        },
+        number: {
+            type: Number
+        },
+        state: {
+            type: String,
+            enum: ["NOT-SENT", "SENT-UNCONFIRMED", "SENT"],
+            default: "NOT-SENT"
```

Comment:
Nice can you explain why did you choose the default to be from "not sent"?

Reply:
Because the first thing the service does is to store the message in the database before sending it, then it tries to send it and in that case its state is updated.